### ISSUE TITLE
Question: Is this some form of reward engineering?

### ISSUE THREAD

**Issue by WorksWellWithOthers** [keon/deep-q-learning#34]
This would break in environments that return the state as more/less than 4 values for unpacking.
1. If not essential can we just remove this?
2. If it's essential, would someone explain why and/or reference the paper for this?
This seems specific to CartPole. I wasn't sure if the implementation's goal was to only solve CartPole. 
 
```x,x_dot,theta,theta_dot = next_state  
r1 = (env.x_threshold - abs(x)) / env.x_threshold - 0.8  
r2 = (env.theta_threshold_radians - abs(theta)) / env.theta_threshold_radians - 0.5  
reward = r1 + r2
```

**Comment by scprotz** [comment]
@WorksWellWithOthers This is indeed a form of reward engineering and is specific to CartPole to turn the returned state into a numeric reward.  Other environments would not need this specifically, and potentially would return a distinct reward already.

**Comment by keon** [comment]
Agreed and fixed in 6e62819. That block was CartPole-only — `env.x_threshold` doesn't exist on other envs (e.g. Acrobot) and the 4-tuple unpack breaks on different state dims. `ddqn.py` now uses the same simple `reward = -10 if done` shaping as `dqn.py`, which is env-agnostic.
